app/src/test/java/rick_and_morty/data/repository/CharacterRepositoryTest.kt

```diff
@@ -21,5 +22,18 @@ class CharacterRepositoryTest {
         classToTest.getCharacters(1)
 
         verify(rickAndMortyApiRemoteDataSource).fetchRickAndMortyData(1)
+
+    }
+
+
+    @Test
+    fun `verify getCharacterDetails has been called`(): Unit = runBlocking {
+
+        given(rickAndMortyApiRemoteDataSource.fetchCharacterDetailsData(1)).willReturn(characterResult)
+
+        classToTest.getCharacterDetails(1)
+
+        verify(rickAndMortyApiRemoteDataSource).fetchCharacterDetailsData(1)
```

Comment:
currently this is an [interaction based test](https://testing.googleblog.com/2013/03/testing-on-toilet-testing-state-vs.html)

let's switch it to state based test.

You can do that by switching the assertion. Instead of verifying `rickAndMortyApiRemoteDataSource` verify the result of `classToTest.getCharacterDetails(1)` is equal to `characterResult`. (and the only way it can be equal to `characterResult` is if it calls the mock with `1` as an argument. 

PS we use Truth as an assertion library https://truth.dev/